### PR TITLE
Added AutoSuggest package and added autoComplete component

### PR DESCRIPTION
**Pull request by anuroop15**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- yes

#### Which User story does this relate to?
-  https://www.pivotaltracker.com/story/show/151300985

#### Brief feature description
- this story allows uses to get address suggestions while typing

#### Does this PR block/effect other stories?
- NA

#### Do any commands needs to be run in order to run the project with this PR?
- npm install 
- npm run dev

#### Code coverage percent diff as a result of this PR?
- rspec -> 73.5%, JS -> 76%

#### JS errors diff as a result of this PR?
-  74

### CODE REVIEW

**Comment by anuroop15** [comment]
Jenkins retest this please

File: config/routes.rb [hunk 1]
```diff
@@ -13,15 +13,17 @@
 
 
   namespace :rfa do
-    resources :a01 do
+        resources :a01 do
```
Comment (ratneshraval) [inline config/routes.rb]: why extra tabs?

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js [hunk 2]
```diff
@@ -55,10 +59,22 @@ export default class AddressCard extends React.Component {
...
           <form>
-            <InputComponent gridClassName='col-md-12' id='street_address'
-              value = {physicalAddressFields.street_address}
-              label='Physical Address:' placeholder=''
-              onChange={(event) => this.onAddressChange(physicalAddressType, 'street_address', event.target.value)} />
+            <div className="col-md-12">
+              <label>Physical Address:</label>
+              <AutoCompleter gridClassName='col-md-12'
+                id='physicalAddress'
+                fieldName={'street_address'}
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: Unnecessary expression {'street_address'}

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js [hunk 3]
```diff
@@ -44,6 +45,9 @@ export default class AddressCard extends React.Component {
     data = data.update(index, x => x.set(key, value))
     this.props.setParentState('addresses', data.toJS())
   }
+  autoFillAddress (autoFilData) {
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: Typo here. should be autoFillData

File: app/controllers/geoservice_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class GeoserviceController < CalsBaseController
+  skip_before_action :authenticate_with_cwds
```
Comment (ratneshraval) [inline app/controllers/geoservice_controller.rb]: Why skip authentication

File: app/javascript/components/common/autoCompleter.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    return suggestion.street_address
+  }
+  onSuggestionsClearRequested () {
+    //this.props.autoFillAddres(suggestion)
+  }
+  onSuggestionSelected (event, {suggestion, suggestionValue, suggestionIndex, sectionIndex, method}) {
+    //this.props.autoFillAddres(suggestion)
+  }
+  onSuggestionsFetchRequested ({ value, reason }) {
+    var url = urlPrefixHelper('/geoservice/')
```
Comment (ratneshraval) [inline app/javascript/components/common/autoCompleter.jsx]: Pass this URL as prop

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js [hunk 6]
```diff
@@ -55,10 +59,22 @@ export default class AddressCard extends React.Component {
...
-              label='Physical Address:' placeholder=''
-              onChange={(event) => this.onAddressChange(physicalAddressType, 'street_address', event.target.value)} />
+            <div className="col-md-12">
+              <label>Physical Address:</label>
+              <AutoCompleter gridClassName='col-md-12'
+                id='physicalAddress'
+                fieldName={'street_address'}
+                addressType={physicalAddressType}
+                value = {physicalAddressFields.street_address}
+                autoFillAddres={this.autoFillAddress.bind(this)}
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: can you change this name from autoFIllAddres to onSelection?

File: app/controllers/geoservice_controller.rb (hunk 4 above)
Comment (anuroop15) [reply app/controllers/geoservice_controller.rb]: removed

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js (hunk 2 above)
Comment (anuroop15) [reply app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: addressed

File: app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js (hunk 3 above)
Comment (anuroop15) [reply app/javascript/rfa_forms/rfa01a_edit_view/addressCard.js]: addressed

File: config/routes.rb (hunk 1 above)
Comment (anuroop15) [reply config/routes.rb]: addressed

File: app/javascript/components/common/autoCompleter.jsx (hunk 5 above)
Comment (anuroop15) [reply app/javascript/components/common/autoCompleter.jsx]: addressed

**Comment by ratneshraval** [comment]
1. Change all vars to let/const. I found 3-4 instance in your files
2. Add propTypes to the autoCompleter.jsx

**Pull request merged**